### PR TITLE
Adds support for multiple schemes

### LINKED ISSUE #6: Support multiple schemes
Modify `@DeepLink` and `@DeepLinks` to also require the `scheme://` in the URI definition, so we can suport more than one scheme in `DeepLinkActivity`.

### PR DESCRIPTION
**Pull request by felipecsl**
Fixes #6

### CODE REVIEW

**Comment by felipecsl** [comment]
@cdeonier

File: README.md [hunk 1]
```diff
@@ -38,7 +38,7 @@ Sometimes you'll have an activity that should handle several kinds of deep links
 `@DeepLinks` annotation to register multiple deep links on an activity:
 
 ```java
-@DeepLinks({"example.com/deepLink/{id}", "example.com/anotherDeepLink"})
+@DeepLinks({"foo://example.com/deepLink/{id}", "foo://example.com/anotherDeepLink"})
```
Comment (cdeonier) [inline README.md]: Maybe include an example where the scheme is optional (ie, omitted).

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java [hunk 2]
```diff
@@ -27,76 +31,75 @@
...
-    this.regex = generateLookupString(hostPath);
+  DeepLinkEntry(String uri, Type type, Class<?> activityClass, String method) {
     this.type = type;
     this.activityClass = activityClass;
     this.method = method;
+    this.parameters = parsePathParameters(uri);
+    this.regex = schemeHostAndPath(uri).replaceAll(PARAM_REGEX, PARAM_VALUE);
   }
 
-  public String getRegex() {
```
Comment (cdeonier) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java]: Not sure why we got rid of the getter/setters here. I understand that it's still accessible as a member variable, but my preference is that it stays accessed through these methods.

File: README.md (hunk 1 above)
Comment (felipecsl) [inline README.md]: My plan was to make the scheme mandatory (even though this is a breaking change) in order to make it a valid URI

Comment (felipecsl) [inline README.md]: ... and also remove ambiguity

File: deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java (hunk 2 above)
Comment (felipecsl) [inline deeplinkdispatch/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkEntry.java]: ok! I'll bring back the getters!

**Comment by cdeonier** [comment]
:+1:

File: README.md (hunk 1 above)
Comment (MichaelEvans) [inline README.md]: If this is a breaking change, is it worth throwing an exception for a missing scheme in the processor?

Comment (felipecsl) [inline README.md]: Compilation will fail if the URI is invalid. It is now being checked by `DeepLinkAnnotatedElement`

Comment (MichaelEvans) [inline README.md]: Ok cool! (Was reading this on mobile and missed that)

**Pull request merged**